all/experiments/writer.py

```
@@ -71,6 +78,70 @@ def _get_step(self, _type):
+    Args:
+        experiment (all.experiments.Experiment): The Experiment associated with the Writer object.
+        agent_name (str): The name of the Agent the Experiment is being performed on
+        env_name (str): The name of the environment the Experiment is being performed in
+        loss (bool, optional): Whether or not to log loss/scheduling metrics, or only evaluation and summary metrics.
+        logdir (str): The directory where run information is stored.
+    '''
+
+    def __init__(self, experiment, agent_name, env_name, loss=True, logdir='runs'):
+        assert COMET_AVAILABLE, "Failed to import comet_ml. CometWriter requires that comet_ml be installed"
+        self.env_name = env_name
+        self._experiment = experiment
+        self._loss = loss
+
+        try:
+            self._comet = Experiment(project_name=env_name)
+        except ImportError as e:
+            print("See https://www.comet.ml/docs/python-sdk/warnings-errors/ for more info on this error.")
+            raise e
+        except ValueError as e:
+            print("See https://www.comet.ml/docs/python-sdk/advanced/#python-configuration for more info on this error.")
+            raise e
+
+        self._comet.set_name(agent_name)
+        self.log_dir = logdir
+        # super().__init__(log_dir=self.log_dir)
+
+    def add_loss(self, name, value, step="frame"):
+        if self._loss:
+            self.add_evaluation("loss/" + name, value, step)
+
+    def add_evaluation(self, name, value, step="frame"):
+        self._comet.log_metric(name, value, self._get_step(step))
+
+    def add_schedule(self, name, value, step="frame"):
+        if self._loss:
+            self._comet.log_parameter(name, value)
+
+    def add_summary(self, name, mean, std, step="frame"):
+        self.add_evaluation(name + "/mean", mean, step)
+        self.add_evaluation(name + "/std", std, step)
+
+    def add_scalar(self, name, value, step="frame"):
+        pass
```

Not ideal to have this silently fail.